rpc/src/retry.rs

```diff
@@ -0,0 +1,148 @@
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use alloy::{
+    providers::{ProviderBuilder, RootProvider},
+    rpc::{
+        client::ClientBuilder,
+        json_rpc::{RequestPacket, ResponsePacket},
+    },
+    transports::TransportError,
+};
+use tower::{retry::Policy, Layer, Service};
+
+#[derive(Debug)]
+pub struct RetryPolicy {
```

Comment:
Maybe we could use [backoff]( https://docs.rs/backoff/latest/backoff/) crate, no need to implement it ourselves?

Reply:
@frisitano Could I open a ticket to port our retry implementation to backoff crate? We may need backoff mechanism in various scenarios/modules, easy with external backoff library. Long term it would probably be easier than to maintain our implementation.

Reply:
I had looked into a few options but they didn't fit into the `tower` stack which `alloy` uses under the hood. This current implementation appears to be the easiest / best solution for the moment. `alloy` are currently working on introducing a retry client upstream so we should migrate to that when it's ready. I've created an issue for this #102. See [here](https://github.com/alloy-rs/examples/pull/95/files) for retry implementation example.

Reply:
> @frisitano Could I open a ticket to port our retry implementation to backoff crate? We may need backoff mechanism in various scenarios/modules, easy with external backoff library. Long term it would probably be easier than to maintain our implementation.

certainly, feel free 👍